Tests/SonarQube.TeamBuild.PreProcessor.Tests/PreProcessorTests.cs

```diff
@@ -81,28 +82,81 @@ public void PreProc_NonEmptySonarRunnerProperties()
             mockPropertiesFetcher.CheckFetcherArguments("my url", "key");
 
             mockRulesetGenerator.AssertGenerateCalled();
             mockRulesetGenerator.CheckGeneratorArguments("my url", "key");
+
+            logger.AssertErrorsLogged(0);
+            logger.AssertWarningsLogged(0);
+        }
+
+        [TestMethod]
+        public void PreProc_CommandLinePropertiesOverrideServerSettings()
+        {
+            // Checks command line properties override those fetched from the server
+            // Arrange
+            string testDir = TestUtils.CreateTestSpecificFolder(this.TestContext);
+
+            string propertiesFile = CreatePropertiesFile(testDir, "my url", "my user name", "my password");
+
+            MockRulesetGenerator mockRulesetGenerator = new MockRulesetGenerator();
+            TestLogger logger = new TestLogger();
+
+            MockPropertiesFetcher mockPropertiesFetcher = new MockPropertiesFetcher();
+            mockPropertiesFetcher.PropertiesToReturn = new Dictionary<string, string>();
+
+            mockPropertiesFetcher.PropertiesToReturn.Add("key1", "server value 1 - should be overridden");
+            mockPropertiesFetcher.PropertiesToReturn.Add("key2", "server value 2 - should be overridden");
+            mockPropertiesFetcher.PropertiesToReturn.Add("key3", "server value 3 - should not be overridden");
+
+            IDictionary<string, string> additionalSettings = new Dictionary<string, string>();
+            additionalSettings.Add("key1", "cmd line value1");
+            additionalSettings.Add("key2", "cmd line value2");
+            additionalSettings.Add("key4", "cmd line value4");
+
+            string configFilePath;
+            using (PreprocessTestUtils.CreateValidLegacyTeamBuildScope("tfs uri", "build uri"))
+            {
+                TeamBuildSettings settings = TeamBuildSettings.GetSettingsFromEnvironment(new ConsoleLogger());
+                Assert.IsNotNull(settings, "Test setup error: TFS environment variables have not been set correctly");
+                configFilePath = settings.AnalysisConfigFilePath;
+
+                TeamBuildPreProcessor preProcessor = new TeamBuildPreProcessor(mockPropertiesFetcher, mockRulesetGenerator);
+
+                // Act
+                preProcessor.Execute(logger, "key", "name", "ver", propertiesFile, additionalSettings);
+            }
+
+            // Assert
+            AssertConfigFileExists(configFilePath);
+            mockPropertiesFetcher.AssertFetchPropertiesCalled();
+
             logger.AssertErrorsLogged(0);
             logger.AssertWarningsLogged(0);
+
+            AnalysisConfig actualConfig = AnalysisConfig.Load(configFilePath);
+            AssertExpectedAnalysisSetting("key1", "cmd line value1", actualConfig);
+            AssertExpectedAnalysisSetting("key2", "cmd line value2", actualConfig);
+            AssertExpectedAnalysisSetting("key3", "server value 3 - should not be overridden", actualConfig);
```

Comment:
Good spot. It's there to check the "command line only" case. I'll update the values to make the purpose of each key clear.